transport-native-io_uring/src/main/java/io/netty/channel/uring/SockaddrIn.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.uring;
+
+import io.netty.util.internal.PlatformDependent;
+
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+
+import static io.netty.util.internal.PlatformDependent.BIG_ENDIAN_NATIVE_ORDER;
+
+final class SockaddrIn {
+    static final byte[] IPV4_MAPPED_IPV6_PREFIX = {
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, (byte) 0xff, (byte) 0xff };
+    private SockaddrIn() { }
+
+    /**
+     *
+     * struct sockaddr_in {
+     *      sa_family_t    sin_family; // address family: AF_INET
+     *      in_port_t      sin_port;   // port in network byte order
+     *      struct in_addr sin_addr;   // internet address
+     * };
+     *
+     * // Internet address.
+     * struct in_addr {
+     *     uint32_t       s_addr;     // address in network byte order
+     * };
+     *
+     */
+    static int writeIPv4(long memory, InetAddress address, int port) {
+        int written = 0;
```

Comment:
nit: any reason for not just incrementing `memory` in this and `writeIPv6`?

Reply:
I did this to be able to `assert` easily. Let's keep it for now